include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,220 @@ struct video_frmival_enum {
+ *		port {
+ *			property = "port_property";
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @return port node associated with @p pid
+ */
+#define DT_INST_PORT_BY_ID(inst, pid)                                                              \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(inst, pid)),                                \
+		(_DT_INST_PORT_BY_ID(inst, pid)), (DT_INST_CHILD(inst, port)))
+
+#define _DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                    \
+	DT_CHILD(DT_INST_PORT_BY_ID(inst, pid), endpoint_##eid)
+
+/**
+ * @brief Get an endpoint node from its id and its parent port id
+ *
+ * Given a device instance number, a port ID and an endpoint ID, return the endpoint node.
+ * It handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_INST_PORT_BY_ID and below. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_INST_ENDPOINT_BY_ID(inst, 0, 0), property)
+ * @endcode
+ *
+ * The above snippet returns @c "endpoint_property" for each of the @c &device listed below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint {
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		port {
+ *			endpoint@0 {
+ *				reg = <0x0>;
+ *				property = "endpoint_property";
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *
+ *				endpoint@0 {
+ *					reg = <0x0>;
+ *					property = "endpoint_property";
+ *				};
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param inst instance number
+ * @param pid port ID
+ * @param eid endpoint ID
+ * @return endpoint node associated with @p eid and @p pid
+ */
+#define DT_INST_ENDPOINT_BY_ID(inst, pid, eid)                                                     \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_ENDPOINT_BY_ID(inst, pid, eid)),                       \
+		(_DT_INST_ENDPOINT_BY_ID(inst, pid, eid)),                                         \
+			(DT_CHILD(DT_INST_PORT_BY_ID(inst, pid), endpoint)))
+
+/**
+ * @brief Get the device node from its endpoint node id.
+ *
+ * Given an endpoint node id, return its device node id.
+ * This handles various ways of how a port and an endpoint could be defined as described in
+ * @ref DT_NODE_BY_ENDPOINT. For example:
+ *
+ * @code{.c}
+ *	DT_PROP(DT_NODE_BY_ENDPOINT(DT_NODELABEL(ep0)), property)
+ * @endcode
+ *
+ * The above snippet returns @c "device_property" in all of the @c ep0 below:
+ *
+ * @code{.dts}
+ *	&device {
+ *		property = "device_property";
+ *
+ *		port {
+ *			ep0: endpoint@0 {
+ *				reg = <0x0>;
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @code{.dts}
+ *	&device {
+ *		property = "device_property";
+ *
+ *		ports {
+ *			port@0 {
+ *				reg = <0x0>;
+ *
+ *				ep0: endpoint@0 {
+ *					reg = <0x0>;
+ *				};
+ *			};
+ *		};
+ *	};
+ * @endcode
+ *
+ * @param ep endpoint node
+ * @return device node of @n
```

Comment:
I am learning Doxygen and [Breathe](https://breathe.readthedocs.io/en/latest/index.html) as I go, but `@n` is the equivalent of `\n`, and this [introduces a newline](https://www.doxygen.nl/manual/commands.html#cmdn). Maybe [`@p ep`](https://www.doxygen.nl/manual/commands.html#cmdp) or [`@c n`](https://www.doxygen.nl/manual/commands.html#cmdc) was meant?

Reply:
I don't know neither :-). I didn't find any place in Zephyr that use `@n` but forgot to ask you. So, what I am supposed to place here to replace `@n` ?

Reply:
I think something like this would work better: `@p` for parameter. There is no `@p n` parameter here.
```suggestion
 * @return device node associated with @p ep
```

Reply:
done